### PR TITLE
fix layout anchors

### PR DESCRIPTION
**Pull request by zlolss**
element disappear when i set anchors={'right':'right'}.

it seems anchors not work properly (according to the documentation).

i fixed it.

### CODE REVIEW

File: docs/source/layout_guide.rst [hunk 1]
```diff
@@ -63,8 +63,7 @@ being relative to the top left we anchor to, say the bottom right. That would lo
 .. code-block:: python
    :linenos:
 
-    button_layout_rect = pygame.Rect(0, 0, 100, 20)
-    button_layout_rect.bottomright = (-30, -20)
+    button_layout_rect = pygame.Rect(30, 20, 100, 20)
```
Comment (MyreMylar) [inline docs/source/layout_guide.rst:66]: Why are you changing the documentation if, as your PR says, you are changing the code to fit the documentation?

File: pygame_gui/core/ui_element.py [hunk 2]
```diff
@@ -389,48 +389,28 @@ def set_anchors(self, anchors: Optional[Dict[str, Union[str, IUIElementInterface
...
+                return True
+            return False
 
+        # drop invalid anchors
+        if anchors is None:
+            anchors = {}
+        self.anchors = {
+            source:target for source, target in anchors.items() if is_valid_anchor_map(source, target)
+            }
+
```
Comment (MyreMylar) [inline pygame_gui/core/ui_element.py:413]: This looks like a refactoring without changing the code. Is that correct?

File: pygame_gui/elements/ui_panel.py [hunk 3]
```diff
@@ -58,16 +58,18 @@ def __init__(self,
...
+                           relative_rect,
+                           manager,
+                           container,
+                           starting_height=starting_height,
+                           layer_thickness=1,
+                           anchors=anchors,
+                           visible=visible,
+                           parent_element=parent_element,
+                           object_id=object_id,
+                           element_id=[element_id])
```
Comment (MyreMylar) [inline pygame_gui/elements/ui_panel.py:72]: Why have you changed this?

File: pygame_gui/core/ui_element.py [hunk 4]
```diff
@@ -686,7 +643,8 @@ def _update_absolute_rect_position_from_anchors(self, recalculate_margins=False)
 
         new_width, new_height = self._get_clamped_to_minimum_dimensions((new_width, new_height))
         if (new_height != self.relative_rect.height) or (new_width != self.relative_rect.width):
-            self.set_dimensions((new_width, new_height))
+            UIElement.set_dimensions(self, (new_width, new_height))
+            #self.set_dimensions((new_width, new_height))
```
Comment (MyreMylar) [inline pygame_gui/core/ui_element.py:647]: why did you change this?

**Comment by MyreMylar** [comment]
I'm not sure what issue you are trying to fix here. The description says:

> element disappear when i set anchors={'right':'right'}.
> it seems anchors not work properly (according to the documentation).

But the PR is changing the documentation for the {'right':'right'} case to something else.

When I test the current main branch the code works like the current documentation. After this PR it looks like all the positioning code is failing the automated tests so I assume it no longer works.

File: docs/source/layout_guide.rst (hunk 1 above)
Comment (zlolss) [reply docs/source/layout_guide.rst:66]: The incident happened when I tried to use anchors without setting up a container.

```python
import pygame
import pygame_gui
from pygame._sdl2 import Window


if __name__=="__main__":
    # define
    fps = 60
    window_size = (1280, 720)

    # init
    pygame.init()
    
    # create window
    win = Window("pygm_player", window_size, resizable=True)
    win_surface = win.get_surface()
    ui_manager = pygame_gui.UIManager(window_size)

    # test gui

    button_layout_rect = pygame.Rect(0, 0, 100, 20)
    button_layout_rect.bottomright = (-30, -20)

    cancel_button = pygame_gui.elements.UIButton(relative_rect=button_layout_rect,
                                      text='pygame-gui.Cancel',
                                      manager=ui_manager,
                                      object_id='#cancel_button',
                                      anchors={
                                               'right': 'right',
                                               'bottom': 'bottom',
                                               })
    
    ok_button = pygame_gui.elements.UIButton(relative_rect=pygame.Rect(0, 0, 100, 20),
                                  text='pygame-gui.OK',
                                  manager=ui_manager,
                                  object_id='#ok_button',
                                  anchors={
                                           'right': 'right',
                                           })


    # eventloop
    running = True
    clock = pygame.time.Clock()
    while running:
        time_delta = clock.tick(fps)/1000.0
        # event pass
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                running = False
            
            ui_manager.process_events(event)

        # draw window
        ui_manager.update(time_delta)
        win_surface.fill(pygame.Color('#333333'))
        ui_manager.draw_ui(win_surface)
        
        # render
        win.flip()

    pygame.quit()
```
When I ran the script above, I only saw one button in the window. 

It behaves like a collection of bugs.

As you can see, things got worse when I tried to modify it.

Over the past two days, I have felt completely exhausted.

Okey，

I am the BUG

This is the first time I use pull requests, and I’m not sure what to do next.

I have reverted to the original version.

- Please close or remove the request. 

After all, to avoid using anchors without a defined container, perhaps I need to add a **`highlighted warning`** on the [Layout Guide](https://pygame-gui.readthedocs.io/en/latest/layout_guide.html) page.
